### PR TITLE
refactor: ignore disabled account while selecting Income Accounts

### PR DESCRIPTION
**Pull request by ruthra-kumar**
Filter disabled accounts while selecting `Income Account`.

### CODE REVIEW

File: erpnext/accounts/doctype/sales_invoice/sales_invoice.js [hunk 1]
```diff
@@ -568,7 +568,7 @@ cur_frm.fields_dict.write_off_cost_center.get_query = function(doc) {
 cur_frm.set_query("income_account", "items", function(doc) {
 	return{
 		query: "erpnext.controllers.queries.get_income_account",
-		filters: {'company': doc.company}
+		filters: {'company': doc.company, "disabled": 0}
```
Comment (deepeshgarg007) [inline erpnext/accounts/doctype/sales_invoice/sales_invoice.js]: While you are doing this, maybe remove usage of cur_frm and do the standard way using frm triggers?

**Comment by bosue** [comment]
Interesting.

While the right one, Expense Account, automatically discards disabled accounts, the left one doesn‘t. Would have to investigate why, but adding it manually should normally be superfluous.

Also, the right one at least filters for Profit and Loss accounts.

Neither one is configured to properly filter for only the relevant Account Type – Income Account and Expense Account, respectively.
One could say esisting sites may not have their accounts properly categorized, okay, so let‘s not backport this addition. But in develop, the proper filters should be set and enforced.

7D3D3100-26FE-493E-B336-A00579E753DD

**Comment by ruthra-kumar** [comment]
@bosue 
Framework implicitly adds `disabled` or `enabled` filter conditions based on doctype meta, granted the search has no custom query in it. https://github.com/frappe/frappe/blob/6ad2e762d325f0b62c2cf9dc4659497a0df9b3dd/frappe/desk/search.py#L154-L157

In this case, `Income Account` uses a custom query but `Expense Account` doesn't. This explains the different behaviour on the two link fields.

**Comment by bosue** [comment]
@ruthra-kumar: Thanks and sorry, I was a bit tired…
If we don‘t need a custom query here, and I don’t think we do, we should avoid it.
We should also filter for Profit and Loss accounts, at the least.
Filtering for Account Type may be a followup not being backported.

**Comment by ruthra-kumar** [comment]
> @ruthra-kumar: Thanks and sorry, I was a bit tired… If we don‘t need a custom query here, and I don’t think we do, we should avoid it. We should also filter for Profit and Loss accounts, at the least. Filtering for Account Type may be a followup not being backported.

Can't avoid Custom query in this case. Check: https://github.com/frappe/erpnext/blob/9a171db97f67f88fad1589016afb4e808571d04c/erpnext/controllers/queries.py#L603C23-L605

**Comment by ruthra-kumar** [comment]
@Mergify backport version-14

**Pull request merged**